org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/BaseConfigurator.java

```diff
@@ -70,6 +76,8 @@ private void configure(ProjectContext context, IProject project, IProgressMonito
         EclipseProject model = lookupEclipseModel(project);
         progress.worked(1);
 
+        persistentModel.gradleVersion(this.gradleVersion);
```

Comment:
Is configure called if init reports an error to the InitializationContext?  Could this.gradleVersion be null in some cases when init has failed/never called?

Reply:
Yes, it is called, although it could only fail if the `BuildEnvironment` can't be loaded (for example for an invalid build). In that case, we already fail earlier, when loading the `EclipseProject` model.